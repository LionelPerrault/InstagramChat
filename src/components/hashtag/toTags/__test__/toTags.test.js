import React from 'react'
import ToTags from '../toTags'
import { create } from 'react-test-renderer'

describe('ToTags Component', () => {
  const comp = str => <ToTags str={str} />

  it('should match snapshot with simple hashtag string', () => {
    const str = 'A very #good morning'
    const tree = create(comp(str)).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot with complex hashtag string', () => {
    const str = '#This is a ##double hashtag'
    const tree = create(comp(str)).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot with mention string', () => {
    const str = 'A very #good ##morning @ghalib, @zayn'
    const tree = create(comp(str)).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot with link string', () => {
    const str =
      'Hey @ghalib, #checkout https://github.com/yTakkar/React-Instagram-Clone-2.0'
    const tree = create(comp(str)).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
