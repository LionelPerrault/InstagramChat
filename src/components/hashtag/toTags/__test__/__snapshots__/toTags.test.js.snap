// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToTags Component should match snapshot with complex hashtag string 1`] = `
<span>
  <Link
    className="hashtag"
    to="/hashtag/This"
  >
    #This 
  </Link>
  is 
  a 
  ##double 
  hashtag 
</span>
`;

exports[`ToTags Component should match snapshot with link string 1`] = `
<span>
  Hey 
  <Link
    className="hashtag"
    to="/profile/ghalib"
  >
    @ghalib, 
  </Link>
  <Link
    className="hashtag"
    to="/hashtag/checkout"
  >
    #checkout 
  </Link>
  <a
    className="hashtag"
    href="https://github.com/yTakkar/React-Instagram-Clone-2.0"
    target="_blank"
  >
    https://github.com/yTakkar/React-Instagram-Clone-2.0 
  </a>
</span>
`;

exports[`ToTags Component should match snapshot with mention string 1`] = `
<span>
  A 
  very 
  <Link
    className="hashtag"
    to="/hashtag/good"
  >
    #good 
  </Link>
  ##morning 
  <Link
    className="hashtag"
    to="/profile/ghalib"
  >
    @ghalib, 
  </Link>
  <Link
    className="hashtag"
    to="/profile/zayn"
  >
    @zayn 
  </Link>
</span>
`;

exports[`ToTags Component should match snapshot with simple hashtag string 1`] = `
<span>
  A 
  very 
  <Link
    className="hashtag"
    to="/hashtag/good"
  >
    #good 
  </Link>
  morning 
</span>
`;
